File: CJMS-SharedServices/Services/lib/components/InterfaceModels/Match.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.initIMatch = void 0;
function initIMatch(instance) {
    const defaults = {
        match_number: '',
        start_time: '',
        end_time: '',
        on_table1: { table: '', team_number: '', score_submitted: false, createdAt: null, updatedAt: null },
        on_table2: { table: '', team_number: '', score_submitted: false, createdAt: null, updatedAt: null },
        complete: false,
        deferred: false
    };
    return {
        ...defaults,
        ...instance
    };
}
exports.initIMatch = initIMatch;

File: server/src/schemas/match.rs
#[derive(JsonSchema, Clone)]
pub struct OnTable {
  table: String,
  team_number: String,
  score_submitted: bool
}

#[derive(JsonSchema, Clone)]
pub struct Match {
  match_number: String,
  start_time: String,
  end_time: String,
  on_table_first: OnTable,
  on_table_second: OnTable,
  complete: bool,
  deferred: bool,
  custom_match: bool
}